pkg/store/google_secret_manager_store_test.go

```diff
@@ -0,0 +1,487 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	secretmanagerpb "cloud.google.com/go/secretmanager/apiv1/secretmanagerpb"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/googleapis/gax-go/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+// MockGSMClient is a mock implementation of secretmanager.Client
+type MockGSMClient struct {
+	mock.Mock
+}
+
+func (m *MockGSMClient) CreateSecret(ctx context.Context, req *secretmanagerpb.CreateSecretRequest, opts ...gax.CallOption) (*secretmanagerpb.Secret, error) {
+	args := m.Called(mock.Anything, req)
+	if args.Get(0) == nil {
+		return nil, args.Error(1)
+	}
+	return args.Get(0).(*secretmanagerpb.Secret), args.Error(1)
+}
+
+func (m *MockGSMClient) AddSecretVersion(ctx context.Context, req *secretmanagerpb.AddSecretVersionRequest, opts ...gax.CallOption) (*secretmanagerpb.SecretVersion, error) {
+	args := m.Called(mock.Anything, req)
+	if args.Get(0) == nil {
+		return nil, args.Error(1)
+	}
+	return args.Get(0).(*secretmanagerpb.SecretVersion), args.Error(1)
+}
+
+func (m *MockGSMClient) AccessSecretVersion(ctx context.Context, req *secretmanagerpb.AccessSecretVersionRequest, opts ...gax.CallOption) (*secretmanagerpb.AccessSecretVersionResponse, error) {
```

Comment:
```suggestion
func (m *MockGSMClient) GetSecretVersion(ctx context.Context, req *secretmanagerpb.AccessSecretVersionRequest, opts ...gax.CallOption) (*secretmanagerpb.AccessSecretVersionResponse, error) {
```

Reply:
AccessSecretVersion is a method of google cloud secret manager library, it have nothing to do with the interface that we're implementing for store

https://github.com/GoogleCloudPlatform/golang-samples/blob/main/secretmanager/access_secret_version.go#L31

Reply:
Ok, my fault.